### DISCUSSION TITLE
Do we still need the `uuid` library?

### DISCUSSION THREAD

**Discussion by max-programming** [uuidjs/uuid#735] [Q&A] [answered]
First off, amazing library and thank you for keeping this alive.

But I recently discovered that `crypto.randomUUID()` generates a random uuid.

So do we still need this library? If we don't, what tools don't support the `randomUUID` method?

**Comment by broofa** [comment, marked as answer]
tl;dr: If all you need is a version 4 UUID (random) and you only care about modern platforms, then `randomUUID()` is the way to go.

`crypto.randomUUID()` is a relatively new standard.  It is now [broadly supported](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID#browser_compatibility), so shouldn't be an issue for most users.

That said ...

One caveat to be aware of is that because it's a new standard it is subject to the "[secure context only](https://blog.mozilla.org/security/2018/01/15/secure-contexts-everywhere)" policy.  So it won't work in non-secure contexts (i.e. web pages served in production via HTTP rather than HTTPS). See [explanation of "secure context"](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts)

`randomUUID()` also only provides "version 4" (random) UUIDs.  If you want one of the other flavors of UUID (there are 5 officially, [soon to be 8](https://www.ietf.org/archive/id/draft-ietf-uuidrev-rfc4122bis-11.html)), you'll still need a 3rd party solution

**Reply by max-programming** [reply]
Thanks for the detailed response ✌️

**Reply by Motii1** [reply]
I would also extend the answer. According to the [documentation](https://nodejs.org/api/crypto.html#determining-if-crypto-support-is-unavailable) it is possible for Node.js to be built without including support for the `node:crypto` module. In such cases, attempting to import from crypto or calling `require('node:crypto')` will result in an error being thrown.

**Reply by broofa** [reply]
@Motii1: `uuid` depends on `crypto` so won't be of much use in that scenario... not unless you explicitly give it a `rng` function to work with.  🤷
